app/controllers/concerns/partner_category_filter.rb

```diff
@@ -27,10 +28,21 @@ def apply_to(query = Partner)
   end
 
   def categories
+    # only return categories that have partners
+    #   where those partners have addresses or service areas
+    #     in the neighbourhoods of the site
+
     @categories ||= Category
                     .joins(:partner_tags)
+                    .left_joins(:partner_tags)
+                    .left_joins(partners: %i[address service_areas])
+                    .where(
+                      '(addresses.neighbourhood_id in (:neighbourhood_ids) OR service_areas.neighbourhood_id in (:neighbourhood_ids))',
+                      neighbourhood_ids: @site_neighbourhood_ids
+                    )
                     .group('tags.id')
-                    .having('count(tag_id) > 0')
+                    .having('count(partner_tags.tag_id) > 0')
+                    .having('(count(addresses.id) > 0 OR count(service_areas.id) > 0)')
```

Comment:
Is this necessary? Like how would they have been included if they didn't have those values? Does the in on line 40 include null matches?

Reply:
i think you are confusing `left joins` with `left outer joins`. `left joins` must have both but `left outer joins` can have one if present. we need to do it this way as partners can have EITHER a service area OR an address.